docs/_articles/en/v3.x/salesforce-graph-engine/rules.md

```diff
@@ -37,10 +37,43 @@ ApexFlsViolationRule detects [Create, Read, Update, and Delete (CRUD) and Field-
 |				 |SOQL queries that use [WITH SECURITY_ENFORCED](https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_classes_with_security_enforced.htm#apex_classes_with_security_enforced)|
 |				 |Lists filtered by [Security.stripInaccessible](https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_classes_with_security_stripInaccessible.htm)|
 
-Have a look at related [FAQ](./en/v3.x/faq/#questions-about-interpreting-apexflsviolationrule-results) to understand the results generated.
+### Interpreting ApexFlsViolationRule Results
+
+Match any violation message that you receive with these cases to understand more about the violation.
+
+*Common Case*
+
+>_Validation-Type_ validation is missing for _Operation-Name_ operation on _Object-Type_ with fields _Comma-Separated-Fields_
+
+Parameter explanation:
+
+* _Validation-Type_: Type of validation to be added. CRUD requires object-level checks, and FLS requires field-level checks.
+
+* _Operation-Name_: Data operation that must be sanitized.
+
+* _Object-Type_: Object on which the data operations happen. If Graph Engine couldn’t guess the object type, you see the variable name or *SFGE_Unresolved_Argument*.
+
+* _Comma-Separated-Fields_: Fields on which the data operation works. If you see _Unknown_ as the only field or as one of the fields, Graph Engine didn't have enough information to guess the fields, and you must determine the unlisted fields manually.
+
+*Additional Clause Case*
+
+> _Validation-Type_ validation is missing for _Operation-Name_ operation on _Object-Type_ with fields _Comma-Separated-Fields_ - Graph Engine couldn't parse all objects and fields correctly. Confirm manually if the objects and fields involved in these segments have FLS checks: _Unknown-Segments_
+
+Same as the common case, but also Graph Engine isn't confident about the object names or field names it detected. You also see this additional clause when your field or object ends with `__r`. In both cases, review the relational field, object, and the unparsed segments to ensure they have the required CRUD/FLS checks. Next, add an [engine directive](./en/v3.x/salesforce-graph-engine/working-with-sfge/#add-engine-directives) to force Graph Engine to ignore this warning in the next run.
+
+*stripInaccessible Warning Case*
+
+The `stripInaccessible` warning is thrown for all `stripInaccessible` checks on READ access type. Graph Engine has no way to ensure that the sanitized value returned by `SecurityDecision` is the value used in the code that follows the check. Confirm the values manually, then add an engine directive to force Graph Engine to ignore this warning in the next run. Alternatively, disable these violations by using the `--rule-disable-warning-violation` flag or setting its corresponding environment variable, SFGE_RULE_DISABLE_WARNING_VIOLATION, to true.
+
+*Internal Error Case*
```

Comment:
@jfeingold35 I updated the internal error to match the now Approved error text. I also updated the explanatory paragraph below. The only thing I'm not clear on is how best to display the [placeholder error message + stacktrace info]. How will you update that in the Messages file? Let me know and I can update that here, too.

Reply:
I'm not sure. I'll probably have the error message inline, and have the stack trace divided by lines.
E.g., something like 
"Error and stacktrace: Some error Message; CodeLocation1
CodeLocation2
CodeLocation3"
I don't think it's necessary to go more than three frames deep for now.